### DISCUSSION TITLE
Setting the periodic extraction interval for variables

### DISCUSSION THREAD

**Discussion by choyesol** [carissalow/rapids#225] [Q&A] [answered]
I wish to express my gratitude for the invaluable assistance I've received through the library you kindly provided for data analysis.

I am aware that it is possible to modify the extraction interval of variables by adjusting TIME_SEGMENTS in the config.yaml file.

From my recent endeavors, it appears that the extraction is currently set at one-week intervals. 
Could you please confirm if it is possible to extract values at intervals of two weeks or even longer?

I am reaching out because I have encountered difficulties in identifying an appropriate method to achieve this.

Thank you for your attention to this matter.

**Comment by jenniferfedor** [comment, marked as answer]
Hi @choyesol, thank you for using RAPIDS! It is possible to extract features in periodic time segments that are two weeks in duration or longer, however, depending on the values you choose for `repeats_on` and `repeats_value`, these time segments are likely to be overlapping. 

For example, adding the following line to `data/external/timesegments_periodic.csv` will create time segments that are two weeks in duration (`length` = `13D 23H 59M 59S`), but they will repeat every week (`repeats_on` = `wday`) on Mondays (`repeats_value` = `1`):

```
label,start_time,length,repeats_on,repeats_value
twoweeks,00:00:00,13D 23H 59M 59S,wday,1
```

<br>

This would produce features extracted on the following time segments for a participant whose data collection began on 3/12/2020 (note that the the `INCLUDE_PAST_PERIODIC_SEGMENTS` option in `config.yaml` was set to `False`):  

```
"local_segment","local_segment_label","local_segment_start_datetime","local_segment_end_datetime"
"twoweeks#2020-03-16 00:00:00,2020-03-29 23:59:59","twoweeks","2020-03-16 00:00:00","2020-03-29 23:59:59"
"twoweeks#2020-03-23 00:00:00,2020-04-05 23:59:59","twoweeks","2020-03-23 00:00:00","2020-04-05 23:59:59"
"twoweeks#2020-03-30 00:00:00,2020-04-12 23:59:59","twoweeks","2020-03-30 00:00:00","2020-04-12 23:59:59"
"twoweeks#2020-04-06 00:00:00,2020-04-19 23:59:59","twoweeks","2020-04-06 00:00:00","2020-04-19 23:59:59"
"twoweeks#2020-04-13 00:00:00,2020-04-26 23:59:59","twoweeks","2020-04-13 00:00:00","2020-04-26 23:59:59"
...
```

In this case, you may wish to exclude the overlapping segments by dropping the alternating rows (i.e., those that correspond to `twoweeks#2020-03-23 00:00:00,2020-04-05 23:59:59` and `twoweeks#2020-04-06 00:00:00,2020-04-19 23:59:59`).  

<br>

If this does not suit the needs of your analysis, another, more flexible, option would be to use event time segments rather than periodic time segments. This option would involve manually defining a series of non-overlapping, two-week-long events for each participant, depending on their data collection start and end datetimes. Please feel free to refer to the [time segments](https://www.rapids.science/1.9/setup/configuration/#time-segments) section of our documentation for more details about configuring event segments.  

Please let us know if you have any additional questions!

**Reply by choyesol** [reply]
@jenniferfedor 
Thank you for your guidance; it has significantly assisted me.

Given a feature extraction period of two weeks, is it feasible to divide the data into weekdays and weekends? To illustrate, could one differentiate between weekdays in week 1 and week 2, and similarly segregate weekends for week 1 and week 2?

**Reply by jenniferfedor** [reply]
Hi @choyesol, 

I'm glad the above information was helpful, and apologies for the delay in getting back to you. It is possible to process features separately for weekdays and weekend days. Unfortunately, however, RAPIDS does not support discontinuous time segments, so it would not be possible to extract features directly over _two weeks_ of weekend days (e.g., Saturday April 13, Sunday April 14, Saturday April 20, and Sunday April 21, 2024) excluding the intervening weekdays (e.g., Monday April 15 through Friday April 19), or conversely to extract features over _two weeks_ of weekdays excluding the intervening weekend days.  

You could extract features separately for weekdays and weekend days each week with the following periodic time segments, which start, respectively, on Mondays and Saturdays at midnight, and end, respectively, on Fridays and Sundays at 11:59:59 pm:

```
label,start_time,length,repeats_on,repeats_value
weekday,00:00:00,4D 23H 59M 59S,wday,1
weekend,00:00:00,1D 23H 59M 59S,wday,6
```

You could then separately aggregate the two sets of weekday and weekend features within each two-week interval of interest. Just a note of caution, however, that the interpretations of those aggregated features may differ from the descriptions in our documentation. Please let us know if you have any other questions!
